packages/vite/src/node/plugins/css.ts

```diff
@@ -1117,30 +1118,37 @@ async function minifyCSS(css: string, config: ResolvedConfig) {
 
 export async function hoistAtRules(css: string) {
   const s = new MagicString(css)
+  const cleanCss = emptyCssComments(css)
+  let match: RegExpExecArray | null
+
   // #1845
   // CSS @import can only appear at top of the file. We need to hoist all @import
   // to top when multiple files are concatenated.
   // match until semicolon that's not in quotes
-  s.replace(
-    /@import\s*(?:url\([^\)]*\)|"([^"]|(?<=\\)")*"|'([^']|(?<=\\)')*'|[^;]*).*?;/gm,
-    (match) => {
-      s.appendLeft(0, match)
-      return ''
-    }
-  )
+  const atImportRE =
+    /@import\s*(?:url\([^\)]*\)|"([^"]|(?<=\\)")*"|'([^']|(?<=\\)')*'|[^;]*).*?;/gm
+  while ((match = atImportRE.exec(cleanCss))) {
+    s.overwrite(match.index, match.index + match[0].length, '', {
+      contentOnly: true
+    })
+    s.appendLeft(0, match[0])
```

Comment:
Does consecutive `appendLeft` end up respecting the order of imports?
We may want to add a comment to help understanding the use of appendLeft and prepend
```suggestion
    // @charset must be the top-first line, using appendLeft for @import 
    // ensures this order is respected  
    s.appendLeft(0, match[0])
```

Reply:
Yeah `appendLeft` was used to preserve import order. `prepend` would otherwise reverse the order. Will put a comment.

Reply:
Oh and `appendLeft` isn't used to make way for `@charset`. It's to only preserve the order like explained above.

Reply:
Added a comment of

> Use `appendLeft` instead of `prepend` to preserve original `@import` order

Let me know if that's fine too.